### PR TITLE
Add home and minigame teleport cooldown timers

### PR DESCRIPTION
**Pull request by Nightfirecat**
Fixes runelite/runelite#814
Closes runelite/runelite#2153

### CODE REVIEW

**Comment by deathbeam** [comment]
Can you link the original issue?

**Comment by Nightfirecat** [comment]
Done, forgot that github won't respect commits trying to close issues/PRs.

File: runelite-client/src/main/java/net/runelite/client/plugins/timers/TeleportWidget.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+public enum TeleportWidget
+{
+	HOME_TELEPORT,
+	MINIGAME_TELEPORT;
+
+	@Nullable
+	static TeleportWidget of(int widgetId)
+	{
+		final ImmutableList<Integer> homeTeleportIds = ImmutableList.of(
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/timers/TeleportWidget.java]: any reason why these lists need to be recreated every time someone calls this method? just make them static final inside the enum

Comment (Nightfirecat) [reply runelite-client/src/main/java/net/runelite/client/plugins/timers/TeleportWidget.java]: Agreed, fixed this.

File: runelite-client/src/main/java/net/runelite/client/plugins/timers/TeleportWidget.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import com.google.common.collect.ImmutableList;
+import javax.annotation.Nullable;
+import net.runelite.api.widgets.WidgetInfo;
+
+public enum TeleportWidget
+{
+	HOME_TELEPORT,
+	MINIGAME_TELEPORT;
+
+	private static final ImmutableList<Integer> homeTeleportIds = ImmutableList.of(
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/timers/TeleportWidget.java]: UPPER_CASE

File: runelite-client/src/main/java/net/runelite/client/plugins/timers/TeleportWidget.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+		WidgetInfo.SPELL_ARCEUUS_HOME_TELEPORT.getId()
+	);
+	private static final ImmutableList<Integer> minigameTeleportIds = ImmutableList.of(
+		WidgetInfo.MINIGAME_TELEPORT_BUTTON.getId()
+	);
+
+	@Nullable
+	static TeleportWidget of(int widgetId)
+	{
+
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/timers/TeleportWidget.java]: remove this empty line

File: runelite-client/src/main/java/net/runelite/client/plugins/timers/TeleportWidget.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+	HOME_TELEPORT,
+	MINIGAME_TELEPORT;
+
+	private static final ImmutableList<Integer> homeTeleportIds = ImmutableList.of(
+		WidgetInfo.SPELL_LUMBRIDGE_HOME_TELEPORT.getId(),
+		WidgetInfo.SPELL_EDGEVILLE_HOME_TELEPORT.getId(),
+		WidgetInfo.SPELL_LUNAR_HOME_TELEPORT.getId(),
+		WidgetInfo.SPELL_ARCEUUS_HOME_TELEPORT.getId()
+	);
+	private static final ImmutableList<Integer> minigameTeleportIds = ImmutableList.of(
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/timers/TeleportWidget.java]: UPPER_CASE and empty line above I guess

File: runelite-client/src/main/java/net/runelite/client/plugins/timers/TeleportWidget.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.timers;
+
+import com.google.common.collect.ImmutableList;
+import javax.annotation.Nullable;
+import net.runelite.api.widgets.WidgetInfo;
+
+public enum TeleportWidget
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/timers/TeleportWidget.java]: remove public

File: runelite-client/src/main/java/net/runelite/client/plugins/timers/TimersConfig.java [hunk 6]
```diff
@@ -33,6 +33,17 @@
 {
 	@ConfigItem(
 		position = 0,
+		keyName = "showHomeMinigameTeleports",
+		name = "Teleport cooldown timers",
+		description = "Configures whether timers for home and minigame teleport cooldowns are displayed"
+	)
+	default boolean showHomeMinigameTeleports()
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/timers/TimersConfig.java]: Any reason why you couldn't just append this to end?

File: runelite-client/src/main/java/net/runelite/client/plugins/timers/TimersPlugin.java [hunk 7]
```diff
@@ -96,12 +97,18 @@
 	private static final String SUPER_ANTIFIRE_DRINK_MESSAGE = "You drink some of your super antifire potion";
 	private static final String SUPER_ANTIFIRE_EXPIRED_MESSAGE = "<col=7f007f>Your super antifire potion has expired.</col>";
 	private static final String SUPER_ANTIVENOM_DRINK_MESSAGE = "You drink some of your super antivenom potion";
+	private static final ImmutableList<Integer> HOME_TELEPORT_ENDING_ANIMATIONS = ImmutableList.of(
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/timers/TimersPlugin.java]: Just Collection<> is fine for the type declaration

File: runelite-client/src/main/java/net/runelite/client/plugins/timers/TeleportWidget.java (hunk 2 above)
Comment (deathbeam) [reply runelite-client/src/main/java/net/runelite/client/plugins/timers/TeleportWidget.java]: Just Collection<> is fine for the type declaration

File: runelite-client/src/main/java/net/runelite/client/plugins/timers/TeleportWidget.java (hunk 4 above)
Comment (deathbeam) [reply runelite-client/src/main/java/net/runelite/client/plugins/timers/TeleportWidget.java]: Just Collection<> is fine for the type declaration

File: runelite-client/src/main/java/net/runelite/client/plugins/timers/TimersConfig.java (hunk 6 above)
Comment (Nightfirecat) [reply runelite-client/src/main/java/net/runelite/client/plugins/timers/TimersConfig.java]: It's a timer that I'd think most people would want to notice more than the others listed, but I don't have a strong opinion on where it should be placed.

File: runelite-client/src/main/java/net/runelite/client/plugins/timers/TeleportWidget.java (hunk 2 above)
Comment (Nightfirecat) [reply runelite-client/src/main/java/net/runelite/client/plugins/timers/TeleportWidget.java]: Fixed casing, and declared as `Collection`

File: runelite-client/src/main/java/net/runelite/client/plugins/timers/TeleportWidget.java (hunk 4 above)
Comment (Nightfirecat) [reply runelite-client/src/main/java/net/runelite/client/plugins/timers/TeleportWidget.java]: Fixed casing, and declared as `Collection`

File: runelite-client/src/main/java/net/runelite/client/plugins/timers/TimersPlugin.java (hunk 7 above)
Comment (Nightfirecat) [reply runelite-client/src/main/java/net/runelite/client/plugins/timers/TimersPlugin.java]: Declared as `Collection`

File: runelite-client/src/main/java/net/runelite/client/plugins/timers/TeleportWidget.java (hunk 5 above)
Comment (Nightfirecat) [reply runelite-client/src/main/java/net/runelite/client/plugins/timers/TeleportWidget.java]: `public` -> package-private

File: runelite-client/src/main/java/net/runelite/client/plugins/timers/TeleportWidget.java (hunk 3 above)
Comment (Nightfirecat) [reply runelite-client/src/main/java/net/runelite/client/plugins/timers/TeleportWidget.java]: :hocho:

File: runelite-client/src/main/java/net/runelite/client/plugins/timers/TimersPlugin.java [hunk 8]
```diff
@@ -96,12 +98,18 @@
 	private static final String SUPER_ANTIFIRE_DRINK_MESSAGE = "You drink some of your super antifire potion";
 	private static final String SUPER_ANTIFIRE_EXPIRED_MESSAGE = "<col=7f007f>Your super antifire potion has expired.</col>";
 	private static final String SUPER_ANTIVENOM_DRINK_MESSAGE = "You drink some of your super antivenom potion";
+	private static final Collection HOME_TELEPORT_ENDING_ANIMATIONS = ImmutableList.of(
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/timers/TimersPlugin.java]: I would probably just use a == x || b == y, its just 2 items anyway, no need for list

Comment (Nightfirecat) [reply runelite-client/src/main/java/net/runelite/client/plugins/timers/TimersPlugin.java]: Done

**Comment by Nightfirecat** [comment]
@deathbeam Fixed, good catch

**Comment by Nightfirecat** [comment]
Image for blog post:

**Pull request merged**